unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs

```diff
@@ -79,6 +79,7 @@ public SkyboxController()
 
             CommonScriptableObjects.isFullscreenHUDOpen.OnChange += OnFullscreenUIVisibilityChange;
             CommonScriptableObjects.isLoadingHUDOpen.OnChange += OnFullscreenUIVisibilityChange;
+            DataStore.i.skyboxConfig.avatarMatProfile.OnChange += OnAvatarMatProfileOnChange;
```

Comment:
SkyboxController is really ugly, but if possible let's avoid accessing to `DataStore.i` and instead inject the DataStore or specifically the `avatarMatProfile` variable.

Reply:
You're right. Since there are several uses of different data stores, I have injected the whole `DataStore`. Done in: https://github.com/decentraland/unity-renderer/pull/5162/commits/5c2b4c32454a3e76677ff5bbebc37d6002f87645